### DISCUSSION TITLE
Community/maintainer chat?

### DISCUSSION THREAD

**Discussion by tboerger** [go-semantic-release/semantic-release#182] [General]
Hi, cool project. I would love to contribute some implementations I would need to replace the node version within my projects.

Is there any chat to have some more interactive conversations? Slack? Discord? Matrix?

Do you plan to support plugins from other orgs within the plugin registry? Or should all plugins be part of the go-semantic-release org?

**Comment by cliedeman** [comment]
Hi @tboerger 

I think you can use any accessible binary with github.com/hashicorp/go-plugin so it doesn't have to be under the org. If you want to create a generic plugin @christophwitzko will gladly create a repo for you. 

I make very light use of this project these days - I will get around to an azure devops plugin at some point. 

What language do you plan to write the plugins in? 

I'm not really a fan of the go plugins. I don't like having dependencies downloaded during ci. But it does allow people to write plugins in any language

**Comment by cliedeman** [comment]
@christophwitzko can you enable discussions and convert this to one

**Comment by tboerger** [comment]
My preferred language is Go. You can avoid dependency downloading by vendoring, but that's something I never do.

I have already seen an npm publishing plugin, so I'm missing plugins to publish pypi and ruby packages. If the org doesn't matter I can also start implementing that on one of my orgs.

Beside that something comparable to semantic-release-replace-plugin to replace versions by pattern in different files would also be great.

**Reply by cliedeman** [reply]
> My preferred language is Go. You can avoid dependency downloading by vendoring, but that's something I never do.

The plugins work differently. The plugin library downloads the archives right before running based on the manifest

> I have already seen an npm publishing plugin, so I'm missing plugins to publish pypi and ruby packages. If the org doesn't matter I can also start implementing that on one of my orgs.

Plugins to publish ruby and python packages make sense to me.
Are there go libraries to do this or would you expect ruby or python to be available? 

> Beside that something comparable to semantic-release-replace-plugin to replace versions by pattern in different files would also be great.

Looking at the weekly downloads it does seem to be a more obscure plugin:
* https://www.npmjs.com/package/semantic-release
* https://www.npmjs.com/package/@semantic-release/npm
* https://www.npmjs.com/package/semantic-release-replace-plugin

I dont think its sustainable to create a replace plugin for each [use case](https://github.com/go-semantic-release/files-updater-helm) though so maybe porting that plugin directly would be better.

**Reply by tboerger** [reply]
> The plugins work differently. The plugin library downloads the archives right before running based on the manifest

Ah, you aren't refering to the go build process, just a missunderstanding.

> Are there go libraries to do this or would you expect ruby or python to be available?

I haven't checked that so far, but I guess it would make sense to shell out for the native package process.

> I dont think its sustainable to create a replace plugin for each [use case](https://github.com/go-semantic-release/files-updater-helm) though so maybe porting that plugin directly would be better.

Yeah a more generic one could make sense. I have started to integrate the node-based semapntic release into two similar pet project orgs where I'm using plugins like npm, ruby, python publishing beside the replace plugin like https://github.com/kleister/kleister-js, https://github.com/kleister/kleister-ruby or https://github.com/kleister/kleister-python.

**Comment by christophwitzko** [comment]
Hi @tboerger, sorry for getting back to you so late.

You can build plugins and host them in your own repositories. The whole plugin registry feature is just for convenience and not required.
I would be super happy if you could help us extend our plugin ecosystem. Happy to also help you with the setup (either in your own organization or in this one).

go-semantic-release can install plugins from any repo by using the following notation:
`--changelog-generator github:<owner>/<repo>`
Just make sure the binaries are attached to the release like this: https://github.com/go-semantic-release/mock-plugin/releases/tag/v1.2.0

Also take a look at the mock plugin: https://github.com/go-semantic-release/mock-plugin

Cheers,
Christoph

**Reply by tboerger** [reply]
Ah, didn't know that you can use the `github:org/repo` notation, that's a great hint. 

Maybe it would also help to add namespaces to the registry which could handle community contributions a little better to avoid name collisions :)
